src/start/RealmScreen.js

```diff
@@ -53,6 +54,15 @@ class RealmScreen extends PureComponent<Props, State> {
 
     const { dispatch } = this.props;
 
+    if (new WhatwgURL(realm).username !== '') {
```

Comment:
`new WhatwgURL(realm)` will throw an error if `realm` is an invalid URL: https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
I just tested this by hardcoding something wrong for 'realm'. I got this:

```
Possible Unhandled Promise Rejection (id: 0):
TypeError: Invalid URL: asdf
```

I don't get a white-screen crash (the error was thrown from a React event handler), but the spinner does go on indefinitely. I think we can isolate `new WhatwgUrl(realm)` into its own try/catch block, and, on this error, do a `setState` similar to this one, with an appropriate error message saying it's an invalid URL.

Then the email can be checked below that.

Reply:
Ah! Initially, I did add a try-catch for the URL constructor. But then I saw that we already validate the URL in the login button using a regular expression - but I never considered that the form could be submitted by simply tapping the enter key from the keyboard, without needing to touch the button.

Reply:
Okay, I just tested this, and I *cant* submit the form if the URL is incorrect - by tapping enter or by tapping the send button.